lib/modules/manager/poetry/extract.ts

```diff
@@ -56,12 +58,15 @@ function extractFromSection(
     }
 
     const pep503NormalizeRegex = regEx(/[-_.]+/g);
-    const packageName = depName
-      .toLowerCase()
-      .replace(pep503NormalizeRegex, '-');
+    let packageName = depName.toLowerCase().replace(pep503NormalizeRegex, '-');
     let skipReason: SkipReason | null = null;
     let currentValue = sectionContent[depName];
     let nestedVersion = false;
+    let datasource = PypiDatasource.id;
+    let lockedVersion: string | null = null;
+    if (packageName in poetryLockfile) {
+      lockedVersion = poetryLockfile[packageName];
+    }
```

Comment:
Was there a need to refactor/move this logic?

Reply:
Unfortunately yes, we change the packageName to for ex `tiangolo/fastapi.git` when depName and the key in the poetryLockfile is `fastapi`.